### PR TITLE
Create test & Purpose for efficiency

### PR DESCRIPTION
**Pull request by chatblanc-ciel**
Please, check the test or adding codes.

Changing:

- add test fn interpolate
- create test crate, part of test function are moved.
- add purpose for fn interpolate

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,7 @@ Cargo.lock
 
 # These are backup files generated by rustfmt
 **/*.rs.bk
+
+#IDE files
+.settings
+.project
```
Comment (OTL) [inline .gitignore:14]: I think IDE related files should be controlled by each user.
Please use gitignore_global.

File: openrr-planner/src/funcs.rs [hunk 2]
```diff
@@ -185,29 +192,4 @@ where
 
 #[cfg(test)]
 mod tests {
```
Comment (OTL) [inline openrr-planner/src/funcs.rs:194]: It seems that `mod tests` is empty. Could you remove this?

File: openrr-planner/tests/mod.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#[macro_use]
+extern crate approx;
+
+use openrr_planner::*;
+
+#[test]
+fn test_funcs() {
+//	use crate::errors::*;
```
Comment (OTL) [inline openrr-planner/tests/mod.rs:8]: We don't use comment out. Please just remove it.

```suggestion
```

File: openrr-planner/tests/mod.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+#[test]
+fn test_interpolate()
+{
+	use trajectory::{CubicSpline, Trajectory};
+
+// parametor for test
+    let  total_duration = 4.0_f64;
+    let  unit_duration = 0.01_f64;
+	let  points = vec![
```
Comment (OTL) [inline openrr-planner/tests/mod.rs:42]: Indent seems broken. Please use `cargo fmt`.

**Comment by OTL** [comment]
Could you fix those issues?

**Comment by chatblanc-ciel** [comment]
I think all problems are cleared.

Please, check it.

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,4 @@ Cargo.lock
 
 # These are backup files generated by rustfmt
 **/*.rs.bk
+
```
Comment (OTL) [inline .gitignore:11]: Could you remove this change?

File: openrr-planner/src/funcs.rs [hunk 6]
```diff
@@ -148,15 +148,22 @@ pub fn interpolate<T>(
...
     for i in 0..points.len() {
         times.push(key_frame_unit_duration * T::from(i)?);
     }
+    /* FIXME:
+     * 	let time = (0_usize..points.len())
+     *					.map( |i| T::from(i)*key_frame_unit_duration)
+     *					.collect::<Vec<f64>>();
+     * Another purpose
+     * I don't compare and I've not know which is more efficiently yet.
+     */
```
Comment (OTL) [inline openrr-planner/src/funcs.rs:162]: I think you can use your method! At least it reduces `mut`.

**Review by OTL — COMMENTED** [review]
One more thing, thank you for great contribution.

File: .gitignore (hunk 5 above)
Comment (chatblanc-ciel) [reply .gitignore:11]: I can remove it.

But, I didn't write. 
When I use rustfmt, is it always written automaticaly?

File: openrr-planner/src/funcs.rs (hunk 6 above)
Comment (chatblanc-ciel) [reply openrr-planner/src/funcs.rs:162]: I expect that the conventional method needs `mut` .
My method is not changed later process so it is not necessary  `mut`.
I think.

---------------------------
I'll update it. Wait a minute, please.

Comment (chatblanc-ciel) [reply openrr-planner/src/funcs.rs:162]: I expect that the conventional method needs `mut` .
My method is not changed later process so it is not necessary  `mut`.
I think.

---------------------------
I'll update it. Wait a minute, please.

File: .gitignore (hunk 5 above)
Comment (chatblanc-ciel) [reply .gitignore:11]: I can remove it.

But, I didn't write. 
When I use rustfmt, is it always written automaticaly?

Comment (chatblanc-ciel) [reply .gitignore:11]: Sorry, I may be understood.

You tell me to remove neither
```
# These are backup files generated by rustfmt
**/*.rs.bk
```
nor new line.
Is this correct?

Comment (OTL) [reply .gitignore:11]: I mean revert the change about this file.
(remove newline)

File: openrr-planner/Cargo.toml [hunk 7]
```diff
@@ -31,3 +31,5 @@ kiss3d = "0.30"
 structopt = "0.3"
 tracing-subscriber = "0.2"
 urdf-viz = "0.24"
+
+approx = "0.4.0"
```
Comment (OTL) [inline openrr-planner/Cargo.toml:35]: And could you move this line to L30?
We are sorting lines alphabetically.

I mean,

```toml
[dev-dependencies]
approx = "0.4.0"
kiss3d = "0.30"
...
```

Comment (chatblanc-ciel) [reply openrr-planner/Cargo.toml]: The example is Useful for me.

Thank you

**Review by OTL — APPROVED** [review]
👍🏼👍🏼👍🏼👍🏼

**Pull request merged**